### PR TITLE
File mask update

### PR DESCRIPTION
**Pull request by leopucci**
This patch solves https://github.com/elastic/beats/issues/11364

### CODE REVIEW

File: libbeat/common/file/rotator.go [hunk 1]
```diff
@@ -310,7 +310,7 @@ func (r *Rotator) purgeOldBackups() error {
 }
 
 func (r *Rotator) purgeOldIntervalBackups() error {
-	files, err := filepath.Glob(r.filename + "*")
```
Comment (michalpristas) [inline libbeat/common/file/rotator.go]: what happens to the other files? like `filebeat.3`? are they ignored?

Comment (leopucci) [reply libbeat/common/file/rotator.go]: Yes. They will not be part of the rotation

Comment (michalpristas) [reply libbeat/common/file/rotator.go]: this fixes an issue with a rotation.
but also can you make sure other files from other rotator are removed as well?
there are multiple strategies how to accomplish this and i'm not sure which one to use

1. keep them there and wait to be cleaned manually
1. on `New` remove backups of opposite rotator 
1. on rotate - rotate in a way that number of log files will be r.max consisting of mixture of previous and current logs sorted correctly (preference of current). so if i have `max==10` and 7 files from current rotator and 3 of previous one, on rotation i will end up with `8+2` in the next step `9+1` and finally with clean `10+0`
1. on rotate find out `modTime` of each file, sort by `modtime` remove all but `r.max` newest

i'm also ok with doing this in a followup as this PR already fixes an issue with missing logs.

Comment (michalpristas) [reply libbeat/common/file/rotator.go]: thinking more about it we have another most valid option here:
- let's get rid of `filename.X` format
- in this case we don't need to pay attention to file format during rotation
- with new format in mind, we need to transform old names to new ones on startup (if any) `filename.X` => `filename.{file.ModTime}` 

@leopucci let me know if this is something you would like to cover in this PR or I should create an additional issue for this so somebody can address this in a followup

Comment (leopucci) [reply libbeat/common/file/rotator.go]: @michalpristas I have created the function to purge the .X filenames on startup. 
Also created a test function. TestIntervalRotationPurgeFileRotationLogs

daily-2009-11-17-
daily-2009-11-18-
daily-2009-11-19-

Names are not ideal, can be better formatted. Please let me know what you think of it
Regards

File: libbeat/common/file/rotator.go [hunk 2]
```diff
@@ -160,6 +160,9 @@ func NewFileRotator(filename string, options ...RotatorOption) (*Rotator, error)
 	if err != nil {
 		return nil, err
 	}
+	if r.intervalRotator != nil {
+		r.purgeOldIntervalOnStartup()
```
Comment (michalpristas) [inline libbeat/common/file/rotator.go]: log the error returned by this func

File: libbeat/common/file/rotator.go [hunk 3]
```diff
@@ -441,3 +444,39 @@ func (r *Rotator) rotateBySize(reason rotateReason) error {
...
+			}
+
+			logPrefix := r.intervalRotator.LogPrefix(r.filename, fi.ModTime())
+
+			err = os.Link(fileName, logPrefix)
+			if os.IsExist(err) {
+				err = os.Link(fileName, logPrefix+strconv.Itoa(index))
+			}
+			if os.IsPermission(err) {
+
```
Comment (michalpristas) [inline libbeat/common/file/rotator.go]: ???

Comment (michalpristas) [reply libbeat/common/file/rotator.go]: also we have a method called safeRename.
you can rename, don't need to create links. also it makes it a bit difficult to properly rotate older files based on timestamp. 
whatever rotator strategy is used it should end up with `filename-{timestamp}` file and code for the `filename.N` should disappear (except the migration[rename] on startup)

Comment (leopucci) [reply libbeat/common/file/rotator.go]: safeRename? can you tell me where it is? I did not found it. 
The link prevents writing over another file, rename does not offer that. 

Until now no major changes on the behavior was made. 
To change the rotation by size filenames will break all tests and will need more attention. 
Can we do this more carefully in a new Pull Request?

**Comment by faec** [comment]
It looks like this PR might have fallen through the cracks -- is this still waiting for a review, or is it stalled on some of the changes? I don't want to tag it for auto-closing if it just needs to be synced and checked in.

**Comment by faec** [comment]
Stale, closing since pings didn't revive it.

**Pull request closed without merging**